### PR TITLE
Adds a (client) component to discover nodes on the network.

### PR DESCRIPTION
**Pull request by balazsracz**
This component can be used by network browsers.

### CODE REVIEW

File: src/openlcb/NodeBrowser.hxx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+ */
+
+#ifndef _OPENLCB_NODEBROWSER_HXX_
+#define _OPENLCB_NODEBROWSER_HXX_
+
+#include <functional>
+
+#include "openlcb/If.hxx"
+
+namespace openlcb {
```
Comment (bakerstu) [inline src/openlcb/NodeBrowser.hxx]: brace style.

File: src/openlcb/NodeBrowser.hxx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+/// This class helps establishing a list of all (live) nodes on the
+/// network. Useful for configuration tools.
+class NodeBrowser {
```
Comment (bakerstu) [inline src/openlcb/NodeBrowser.hxx]: brace style.

File: src/openlcb/NodeBrowser.hxx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    ~NodeBrowser();
+    
+    /// Requests a pong from every live node. This function will return
+    /// immediately, then callbacks will be called for each reply that arrives.
+    void refresh();
+
+private:
+    /// Helper class to register in the dispatcher. Incomng response messages
+    /// will be routed to this object.
+    class VerifiedHandler : public MessageHandler {
```
Comment (bakerstu) [inline src/openlcb/NodeBrowser.hxx]: brace style.

File: src/openlcb/NodeBrowser.hxx [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    public:
+        VerifiedHandler(NodeBrowser* parent);
```
Comment (bakerstu) [inline src/openlcb/NodeBrowser.hxx]: Missing Doxygen comments on this and other class members.

File: src/openlcb/NodeBrowser.hxx (hunk 1 above)
Comment (balazsracz) [reply src/openlcb/NodeBrowser.hxx]: done

File: src/openlcb/NodeBrowser.hxx (hunk 3 above)
Comment (balazsracz) [reply src/openlcb/NodeBrowser.hxx]: done

File: src/openlcb/NodeBrowser.hxx (hunk 4 above)
Comment (balazsracz) [reply src/openlcb/NodeBrowser.hxx]: done

**Pull request merged**